src/util/amount.rs

```diff
@@ -1078,16 +1110,16 @@ pub mod serde {
             "i"
         }
         fn ser_sat_opt<S: Serializer>(self, s: S) -> Result<S::Ok, S::Error> {
-            s.serialize_some(&self.as_sat())
+            s.serialize_some(&self.to_sat())
         }
         fn ser_btc_opt<S: Serializer>(self, s: S) -> Result<S::Ok, S::Error> {
-            s.serialize_some(&self.as_btc())
+            s.serialize_some(&self.to_btc())
         }
     }
 
-    pub mod as_sat {
+    pub mod to_sat {
```

Comment:
Do these modules really need renaming? Their naming is unrelated to the convention of conversion functions, isn't it?

Without this rename, the patch set would be backwards-compatible thanks to the deprecations..

Reply:
```rust
#deprecate("use <newname> instead")
pub mod as_sat {
  pub use super::to_sat::*;
}
```

could make rename less painful.

Having said that, I don't find either `to_sat` or `as_sat` great names. :shrug:

Reply:
This is sloppy work, no excuse. I agree with @thomaseizinger that the module name is unrelated so should not be part of this PR. Apologies for wasting your time.

Reply:
I think `as_sat` is a good name for serde implementation using satoshi representation.